### ISSUE TITLE
Invoker waiting for hardcoded 1 second

### ISSUE THREAD

**Issue by kpavel** [pywren/pywren-ibm-cloud#303]
Is this sleep(1) mandatory?
https://github.com/pywren/pywren-ibm-cloud/blob/master/pywren_ibm_cloud/invoker.py#L308

If the sleep mandatory, can it be not hardcoded, but based on parameters, same as in wait_storage:
https://github.com/pywren/pywren-ibm-cloud/blob/master/pywren_ibm_cloud/wait/wait_storage.py#L92

**Comment by gilv** [comment]
@JosepSampe do you know why we have this sleep?

**Comment by JosepSampe** [comment]
Invoker is completely async, first it uses threads to spawn the firsts X function activations. The sleep gives some CPU time to the invoker to do the invocation instead of immediately annoying it with other tasks due the Python Global Interpreter Lock. 

@kpavel It is not mandatory but I think recommended. Are you seeing any issue with it? Maybe I forgot to take into account some use case where this sleep could be annoying. Could you post your reasons to make it configurable?

**Comment by kpavel** [comment]
@JosepSampe For me, the only reasons to make it optional/configurable is to make pywren.map() faster. I been doing some benchmarking for our usecase where we need the pywren to execute as fast as possible and noticed this sleep. From my observations making it 0.1 has no impact on actions invocation and execution. It just improves overall map() time from ~1.15sec to ~0.15.

**Comment by JosepSampe** [comment]
Thanks @kpavel. Previous to add this sleep (in previous Pywren versions), invocation phase using threads was sync, this means you had to wait the invoker to finish. Normally, invoke 1000 functions takes 3 seconds, so after changing the invoker to async and adding this sleep of 1 second, the map() became even to 2 seconds faster than before. It is true that if you invoke <333 functions the map() becomes slower than before.

I added this some time ago, and I remember some sort of benefit of having this sleep here as I stated in the previous post. However, as you said, maybe 0.1 seconds is enough, or maybe now we don't need this sleep anymore. I will run some tests to verify it.
